corehq/apps/locations/resources/v0_6.py

```diff
@@ -35,4 +39,68 @@ def dehydrate(self, bundle):
             bundle.data['parent_location_id'] = ''
         bundle.data['location_type_name'] = bundle.obj.location_type.name
         bundle.data['location_type_code'] = bundle.obj.location_type.code
+        bundle.data['location_data'] = bundle.obj.metadata
+        return bundle
+
+    def obj_create(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        if 'name' not in bundle.data:
+            raise BadRequest("'name' is a required field.")
+        if SQLLocation.objects.filter(domain=domain, site_code=bundle.data['site_code']).exists():
+            raise BadRequest("Location on domain with site code already exists.")
+        bundle.obj = SQLLocation(domain=domain)
+        self._update(bundle)
         return bundle
+
+    def obj_update(self, bundle, **kwargs):
+        try:
+            bundle.obj = SQLLocation.objects.get(location_id=kwargs['location_id'])
+        except SQLLocation.DoesNotExist:
+            raise BadRequest("Could not find location with given ID.")
+        if bundle.obj.domain != kwargs.pop('domain'):
```

Comment:
Out of curiosity, why `pop` domain here?  Also I think you can move this to the `.get` above and save a line